lusSTR/filter_settings.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python
+#
+# -----------------------------------------------------------------------------
+# Copyright (c) 2020, Battelle National Biodefense Institute.
+#
+# This file is part of lusSTR (http://github.com/bioforensics/lusSTR)
+# and is licensed under the BSD license: see LICENSE.txt.
+# -----------------------------------------------------------------------------
+
+import json
+import lusSTR
+import numpy as np
+import pandas as pd
+from pkg_resources import resource_filename
+
+
+def get_filter_metadata_file():
+    return resource_filename('lusSTR', 'filters.json')
+
+
+with open(get_filter_metadata_file(), 'r') as fh:
+    filter_marker_data = json.load(fh)
+
+
+def filters(data_order, locus, total_reads, datatype):
+    metadata = filter_marker_data[locus]
+    if len(data_order) == 1:
+        if thresholds('Detection', metadata, total_reads, data_order['Reads'][0])[1] is False:
+            data_order = pd.DataFrame()
+        elif thresholds('Analytical', metadata, total_reads, data_order['Reads'][0])[1] is False:
+            data_order[['allele_type', 'perc_noise']] = ['noise', 1.0]
+        elif thresholds('Analytical', metadata, total_reads, data_order['Reads'][0])[1] is True:
+            data_order['allele_type'] = 'real_allele'
```

Comment:
I think this function is my biggest remaining source of concern. It's doing a lot of things, and I'm wondering if it can be decomposed into 2-4 smaller functions: one that handles `len(data_order) == 1`, one that handles `datatype == "ce"`, and so on.

Also, `data_order` is the allele read count information, right? Another pass over this code to revisit variable names would probably improve legibility quite a bit.

Reply:
ok I'll take a look

Reply:
I did some refactoring and renaming variables to hopefully make things a little more clear. Let me know when you get a chance to look @standage and if it's up to your standard 😆

Reply:
Lol, just so we're clear, I don't think *I* code up to "my standard" very frequently. I hope this review has been helpful, or at least that it increases the likelihood that someone (possibly one of us) looking at this code 3 years will make sense of it!

Reply:
I'm kidding! Hence the laughing face. The Standage standard is great!

Reply:
And much needed for my code 😄

Reply:
The code is much clearer now. Thanks for humoring me!

Reply:
Thank you for your help!!